app/graphql/mutations/cart/add_to_cart.rb

```diff
@@ -0,0 +1,29 @@
+module Mutations
+    module Cart
+        class AddToCart < BaseMutation
+            argument :input, Types::UpdateCartInputType, required: true
+
+            field :cart, [Types::CartItemType], null: false
+            
+            def resolve(input:)
+                cart = context[:cart]
+                product_id = input.product_id
+                quantity = input.quantity
+
+                cart[product_id] = 0 if !cart[product_id]
```

Comment:
Nit: Probably don't need this since you initialized the cart with `{ |h, k| h[k] = 0}` ??

Reply:
I thought so too but for some reason I was still getting an error when trying to `+=` the quantity when the cart was empty, I'll play around with it and see if I can find out why